src/Symfony/Component/Messenger/Tests/Transport/AmqpExt/ConnectionTest.php

```diff
@@ -118,6 +131,33 @@ public function testSetsParametersOnTheQueueAndExchange()
         $connection->publish('body');
     }
 
+    public function testSetsInvalidArgumentsOnTheQueueFails()
+    {
+        $dsn = 'amqp://localhost/%2f/messages?'.
+            'queue[arguments][x-dead-letter-exchange]=dead-exchange&'.
+            'queue[arguments][x-message-ttl]=not-a-number&'
+        ;
+
+        $this->expectsException(\InvalidArgumentException::class);
+        $this->expecteExceptionMessage('Invalid value provided for queue argument x-message-ttl: \'not-a-number\' is not a valid integer value');
+        Connection::fromDsn($dsn);
+    }
+
+    public function testSetsInvalidArgumentsOnTheQueueOptionsFails()
```

Comment:
[minor] This method name could be more explicit like as: `testFromDsnWithNullOnQueueArguments`.

Also if both new test cases have the same kind of assertion you could use `@dataProvider` annotation to remove duplicated code, then the method name could become: `testFromDsnWithInvalidValueOnQueueArguments`.